@xen-orchestra/backups/writers/_DeltaBackupWriter.js

```diff
@@ -224,3 +229,7 @@ exports.DeltaBackupWriter = class DeltaBackupWriter {
     // TODO: run cleanup?
   }
 }
+
+Object.assign(DeltaBackupWriter.prototype, abstractExportWriter)
```

Comment:
After reflection, maybe we should go the other way: having delta/full defined as interfaces (mixins with default implementations) and having the true class hierarchy be for backups vs replications, because they are much more prone to share states.

What do you think?

Reply:
If backups/replications will share more logic than delta/full, yes, it will be better to be classes.

Reply:
Sorry, I changed my mind after thinking about using mixin classes.

`AbstractDeltaWriter` and `AbstractFullWriter` should be the true hierarchy because they are defining the interface to use and can therefore be useful for `instanceof`.

`AbstractExportWriter` can now be used to share implementations when used as a mixin.

Does that make sense?

Reply:
Yes, I agree.